azure/templates/virtualMachine.json

```diff
@@ -61,7 +62,7 @@
           },
           "dataDisks": [
             {
-              "lun": 0,
+              "lun": "[parameters('vmSetting').azDisk]",
```

Comment:
This field requires a integer. But you are passing in a string. Please make it consistent.